Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -34,38 +32,43 @@ bool LogFileReader::readAll(const QString& fileName)
     {
         QString line = input.readLine();
         BatteryData batteryData;
-        if (!parseLine(line, batteryData))
+        if (!parseLine(line, batteryData))  //! taken before parseline out
         {
             qDebug() << "Error while parsing" << line;
-            // return false;
+            return false;
         }
         else
         {
-            // This is how to send out a signal in QT using the emit keyword.
-            // This line notifies the classes listening to this signal
-            // that battery data has been received.
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
-/* File input is a csv file in the format of hh:mm:ss:zzz, voltage, current.
- * Negative current values denote a charging battery.
- * Need to implement error checking for the correct number of values and
- * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    bool ok;
     QStringList sections = line.split(BATDATA_DELIMITER);
-
+    if(sections.size() != 3)
+    {
+        return false;
+    }  
+    if(sections.contains("[Aa-Zz]")) //make sure it doesnt contain alphabetical characters
```

Comment:
I don't think the function can handle expressions like this. If it does, can I get a link to the docs that describes this? I see this in the docs 
http://doc.qt.io/qt-5/qstringlist.html#contains